adagrams/game.py

```diff
@@ -1,11 +1,95 @@
+# Wave 1 
+import random
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
 def draw_letters():
-    pass
+    letters = []
+    all_letter_pool = []
+    for letter in LETTER_POOL:
+        letter_freq = LETTER_POOL[letter]
+        all_letter_pool += letter * letter_freq
+
+    for i in range(10):
+        letter_choice = random.choice(all_letter_pool)
+        letters.append(letter_choice)
+        all_letter_pool.remove(letter_choice)
```

Comment:
Rather than using a list of the letters in the hand, could we build a helper data structure (like a dictionary) that could let us look up if a letter was part of a user's hand?

If the helper dictionary had keys that were the letters and the values were the number of letters in the hand, then you could leverage the fact that key look up in dictionaries is a constant O(1) operation.

You can check if key (the letter in the hand) in helper_dictionary, which has constant look up time. Something like this:
```
def draw_letters():
    letter_pool_copy = copy.deepcopy(LETTER_POOL)
    letters = []

    while len(letters) < 10:
        letter = random.choice(list(letter_pool_copy.keys()))
        if letter_pool_copy[letter] > 0:
            letters.append(letter)
            letter_pool_copy[letter] -= 1

    return letters
```
Additionally, instead of using the remove method, you could decrement the value in the dictionary to keep track of how many letters have been played. Recall that the remove method on a list also has linear run time because it could have to iterate over the entire list before finally removing letter.